### PR TITLE
Death and jumping animations, refactored lots

### PR DESCRIPTION
**Pull request by josepharhar**
#148 #150 #60 #99

### CODE REVIEW

**Comment by bbeck13** [comment]
Fixed some of your logic 
CR 👍 QA 👍

File: src/game_state/GameState.cpp [hunk 1]
```diff
@@ -155,7 +155,14 @@ void GameState::SetPlayingState(PlayingState playing_state) {
       }
       break;
     case GameState::PlayingState::FAILURE:
+      if (music->getStatus() == sf::SoundSource::Status::Playing) {
+        music->stop();
```
Comment (bbeck13) [inline src/game_state/GameState.cpp]: The music should stop when the player fails

File: src/game_updater/GameUpdater.cpp [hunk 2]
```diff
@@ -71,9 +76,10 @@ void GameUpdater::Update(std::shared_ptr<GameState> game_state) {
     return;
   }
 
-  // check to see if the music should start on this tick
-  std::shared_ptr<sf::Music> music = game_state->GetLevel()->getMusic();
-  if (game_state->GetMusicStartTick() == game_state->GetElapsedTicks()) {
+    // check to see if the music should start on this tick
+    std::shared_ptr<sf::Music> music = game_state->GetLevel()->getMusic();
+  if ((game_state->GetMusicStartTick() >= game_state->GetElapsedTicks()) &&
+      (music->getStatus() != sf::SoundSource::Status::Playing)) {
```
Comment (bbeck13) [inline src/game_updater/GameUpdater.cpp]: A bit of paranoia here

**Review by bbeck13 — DISMISSED** [review]

**Comment by alexottoboni** [comment]
You die, if you hit the ground too fast. 👎

**Comment by josepharhar** [comment]
@alexottoboni I believe that I fixed the death on hitting the ground too fast. #150 
@bbeck13 @Matthew-Stewart are you guys still having any sfml issues on this branch?

**Comment by Matthew-Stewart** [comment]
Not sure if I had sfml issues before, but everything seems to be working

File: src/game_updater/PlayerUpdater.cpp [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                                            player->GetWheelRotationSpeed());
+
+  if (current_animation == Player::Animation::FAILURE) {
+    // go crazy
+    player->SetRotationAngle(player->GetRotationAngle() + 0.3f);
+  }
+
+  if (current_animation & Player::ANIMATION_AERIAL_TILT_BIT) {
+    // set player rotation based on y velocity
+    player->SetRotationAngle(std::atan(player->GetYVelocity() * 4));
```
Comment (bbeck13) [inline src/game_updater/PlayerUpdater.cpp]: It looks a little jaring when the player lands from a long fall can you try this
```cpp
float vel = std::atan(player->GetYVelocity()*4;
if (vel < 0)
  player->SetRotationAngle(std::min(0.872665, vel);
else
    player->SetRotationAngle(std::max(-0.872665, vel);
```
0.872665 is about 50 degrees y The idea here is that the player should never go past +-50 degrees.

**Review by bbeck13 — CHANGES_REQUESTED** [review]
refactors look ok overall I'm able to follow the style and the changes you made, game runs fine 👌

File: src/game_updater/PlayerUpdater.cpp (hunk 3 above)
Comment (josepharhar) [reply src/game_updater/PlayerUpdater.cpp]: Done, not sure if i can tell the difference though

File: src/game_updater/PlayerUpdater.cpp [hunk 4]
```diff
@@ -15,6 +15,7 @@
 #include "CollisionCalculator.h"
 
 #define COLLISION_WIDTH 0.420f
+#define MAX_AERIAL_ROTATION_ANGLE 0.872665f
```
Comment (bbeck13) [inline src/game_updater/PlayerUpdater.cpp]: Change this to .67 it looks a lot better

**Review by bbeck13 — CHANGES_REQUESTED** [review]

File: src/game_updater/PlayerUpdater.cpp (hunk 4 above)
Comment (josepharhar) [reply src/game_updater/PlayerUpdater.cpp]: Done, I can tell the difference now :ok_hand:

**Review by bbeck13 — APPROVED** [review]

**Comment by bbeck13** [comment]
CR 👍 QA 💺

**Pull request merged**